### PR TITLE
Configure chatbot fields, intructions on AI chat levels

### PR DESCRIPTION
**Pull request by bencodeorg**
Allows levelbuilders to edit/store configuration for `aichat` levels -- a system prompt, student-facing description, and a bot title.

image

I also added the ability for levelbuilders to edit long instructions, which I believe is the set of instructions we'd want to be able to edit (I referred to [this Javalab PR](https://github.com/code-dot-org/code-dot-org/pull/39631)). I also checked in with Amy to see if there were any additional fields we'd want to have, which she didn't think were necessary at the moment.

I excluded the image upload aspect of this feature from this PR -- after chatting with Amy, it sounds like there is a bit of uncertainty about how we'll configure chatbot avatars (ie, will the design team need to manage avatars for each chatbot?) We can add this at a later date if needed.

## Links

- jira ticket: [SL-996](https://codedotorg.atlassian.net/browse/SL-996)

## Testing story

Tested manually that editing these fields were a) available on revisiting the edit page, b) cloned when I cloned the level, and c) were stored to the level config file as expected:

image

### CODE REVIEW

File: dashboard/app/models/levels/aichat.rb [hunk 1]
```diff
@@ -24,14 +24,11 @@
 #  index_levels_on_name       (name)
 #
 
-# TODO: Update this level to account for the params we actually need.
 class Aichat < Level
   serialized_attrs %w(
-    project_template_level_name
```
Comment (bencodeorg) [inline dashboard/app/models/levels/aichat.rb:30]: These don't seem like attributes we'll need for these levels, but I could be wrong? Easy enough to re-add if we do.

Comment (ebeastlake) [reply dashboard/app/models/levels/aichat.rb:30]: Yeah, Brendan helped confirm when I originally asked/double-checked that any of the serialized_attrs can changed later: https://github.com/code-dot-org/code-dot-org/pull/52538/files#r1260226028

**Review by ebeastlake — APPROVED** [review]
Might want a second 👍 from someone who knows what they're doing with levelbuilder, but this looks good to me!

File: dashboard/app/views/levels/editors/_aichat.html.haml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 = render partial: 'levels/editors/fields/aichat_fields', locals: {f: f}
-
+= render partial: 'levels/editors/fields/instructions_area', locals: {f: f}
```
Comment (fisher-alice) [inline dashboard/app/views/levels/editors/_aichat.html.haml:2]: No video levels needed - correct?

File: dashboard/app/views/levels/editors/fields/_instructions.haml [hunk 3]
```diff
@@ -2,6 +2,6 @@
   Instructions
 
 #instructions.in.collapse
-  = render partial: 'levels/editors/fields/short_instructions', locals: {f: f} unless (@level.is_a?(Applab) || @level.instance_of?(Gamelab) || @level.is_a?(Weblab) || @level.is_a?(ExternalLink) || @level.is_a?(FreeResponse) || @level.is_a?(FrequencyAnalysis) || @level.is_a?(PublicKeyCryptography)|| @level.is_a?(Vigenere) || @level.is_a?(Javalab))
+  = render partial: 'levels/editors/fields/short_instructions', locals: {f: f} unless (@level.is_a?(Applab) || @level.instance_of?(Gamelab) || @level.is_a?(Weblab) || @level.is_a?(ExternalLink) || @level.is_a?(FreeResponse) || @level.is_a?(FrequencyAnalysis) || @level.is_a?(PublicKeyCryptography)|| @level.is_a?(Vigenere) || @level.is_a?(Javalab)) || @level.is_a?(Aichat)
```
Comment (fisher-alice) [inline dashboard/app/views/levels/editors/fields/_instructions.haml:5]: What does adding `Aichat` to this line mean - that the AI chat's instructions does NOT toggle between short and long and just long instructions are displayed? Just curious.

File: dashboard/app/views/levels/editors/_aichat.html.haml (hunk 2 above)
Comment (bencodeorg) [reply dashboard/app/views/levels/editors/_aichat.html.haml:2]: I think so -- I chatted with Amy BW about that, who said videos in their own levels would be fine. Is there something about video configuration that's permanent (ie, if we wanted to add it later we wouldn't be able to)?

File: dashboard/app/views/levels/editors/fields/_instructions.haml (hunk 3 above)
Comment (bencodeorg) [reply dashboard/app/views/levels/editors/fields/_instructions.haml:5]: tbh I don't know exactly -- I followed a lot of the wisdom in this PR from Dani, who suggested that long instructions were the way (and that short instructions were for "CSF style levels":

https://github.com/code-dot-org/code-dot-org/pull/39631

File: dashboard/app/views/levels/editors/_aichat.html.haml (hunk 2 above)
Comment (fisher-alice) [reply dashboard/app/views/levels/editors/_aichat.html.haml:2]: I don't know about video configuration that's permanent, but videos in their own levels sounds good!

File: dashboard/app/views/levels/editors/fields/_instructions.haml (hunk 3 above)
Comment (fisher-alice) [reply dashboard/app/views/levels/editors/fields/_instructions.haml:5]: Interesting - didn't know short instructions were only meant for CSF style levels. Thanks!

**Review by fisher-alice — APPROVED** [review]
Looks good - thanks for answering my questions.

**Pull request merged**